src/data/markdown/docs/05 Examples/01 Examples/09 data-uploads.md

```diff
@@ -86,3 +86,52 @@ and sent.
 
 - [open(filePath, [mode])](/javascript-api/init-context/open-filepath-mode)
 - [http.file(data, [filename], [contentType])](/javascript-api/k6-http/file-data-filename-contenttype)
+
+
+## Advanced multipart request
+
+The previous multipart request example has some limitations:
+
+- It's not possible to assemble the parts in a specific order, because of the
+  unordered nature of JS objects when they're converted to Golang maps, which k6 uses internally.
```

Comment:
nit: This makes it sound like JS objects don't have an order to their keys ... but they do. I am pretty sure that even in goja that works. The problem is that they are converted to golang maps, which don't, but I am not certain this is made clear from the wording here. I tried fixing it ... but with no success :(

Reply:
AFAIK key order was unspecified in <=ES5, and browsers did their own implementation. Even in >=ES6 [the situation is messy](https://stackoverflow.com/a/30919039/96213), and integer keys are ordered numerically, while others are ordered by insertion.

I'm not sure how goja handles this internally, but for our purposes (and I'd argue in general for JS) it's clearer to not rely on objects for order, and use something like `Map` which explicitly supports it.

I thought the phrasing here is clear that it's an issue when they're converted to Golang maps and didn't want to go into any more detail. So suggestions for how to change it is welcome.

Reply:
It just has the property names as a list  ;) 
https://github.com/dop251/goja/blob/f5884268f0c0a0eb94423dcf7c2eda14907375e6/object.go#L191-L199

as I said, this is a nitpick. Unless somebody else comes up with better wording, as I can't myself as well, it is fine to just merge it as is.